core/arch/x86_64/mm/core_mmu.c

```diff
@@ -1967,14 +1976,46 @@ static int arch_mmu_map(vaddr_t vaddr, paddr_t paddr,
 	range.start_paddr = paddr;
 	range.size = count;
 
-	return x86_mmu_map_range(virt_to_phys((void *)&g_pml4[0]),
-								&range, flags);
+	return x86_mmu_map_range(virt_to_phys((void *)&g_thread_pml4[id][0]),
+								&range, flags, id);
+}
+
+static void init_kernal_mmu_table(void)
+{
+    uint8_t thread_id;
+
+	for (thread_id=1; thread_id<CFG_NUM_THREADS; thread_id++) {
+		memcpy(g_thread_pdp[thread_id], g_thread_pdp[0], sizeof(uint64_t) * NO_OF_PDP_ENTRIES);
+
+		g_thread_pml4[thread_id][0] = (g_thread_pml4[0][0] & X86_FLAGS_MASK) |
+			((uint64_t)&g_thread_pdp[thread_id][0] & X86_PG_PA_FRAME);
+	}
+}
+
+static void init_user_ta_mmu_table(struct tee_mmap_region *ta_ram_mm)
+{
+	int ret;
+	uint8_t thread_id;
+	size_t size = ((ta_ram_mm->size) / CFG_NUM_THREADS);
+
+	core_mmu_get_user_va_range(&(ta_ram_mm->va), NULL); // currently, 0xF0000000
+
+	for (thread_id=0; thread_id<CFG_NUM_THREADS; thread_id++) {
```

Comment:
Previous part is building for kernel space mapping. Build thread 0 firstly, and spread to other thread in init_kernal_mmu_table(). 
This function, init_user_ta_mmu_table() is building user space mapping from thread 0, because each thread has different mapping. 
But now, I don't want to build user space mapping in the boot time, want to align with ARM when open session.
Just put function here, will not use.